handler/src/main/java/io/netty/handler/ssl/SslUtils.java

```diff
@@ -71,6 +74,54 @@
      */
     static final int NOT_ENCRYPTED = -2;
 
+    static final String[] DEFAULT_CIPHER_SUITES = new String[] {
+        // GCM (Galois/Counter Mode) requires JDK 8.
+        "TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
+        "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
+        "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
+        "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA",
+        // AES256 requires JCE unlimited strength jurisdiction policy files.
+        "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA",
+        // GCM (Galois/Counter Mode) requires JDK 8.
+        "TLS_RSA_WITH_AES_128_GCM_SHA256",
+        "TLS_RSA_WITH_AES_128_CBC_SHA",
+        // AES256 requires JCE unlimited strength jurisdiction policy files.
+        "TLS_RSA_WITH_AES_256_CBC_SHA"
+    };
+
+    /**
+     * Add elements from {@code names} into {@code enabled} if they are in {@code supported}.
+     */
+    static void addIfSupported(Set<String> supported, List<String> enabled, String... names) {
+        for (String n: names) {
+            if (supported.contains(n)) {
+                enabled.add(n);
+            }
+        }
+    }
+
+    static void useFallbackCiphersIfDefaultIsEmpty(List<String> defaultCiphers, Iterable<String> fallbackCiphers) {
+        if (defaultCiphers.isEmpty()) {
+            for (String cipher : fallbackCiphers) {
+                if (cipher.startsWith("SSL_") || cipher.contains("_RC4_")) {
```

Comment:
`SSL_` cipher may be returned by the JDK in `getSupportedCiphers` however the JDK should accept the `TLS_` ciphers. The reason we exclude `SSL_` is to prevent duplicate entries, and some JDKs may not support this custom naming scheme (but all should support the `TLS_` naming scheme).

There may be some older cipher names in the JSSE which have the `SSL_` prefix (established before the `TLS_` ... e.g. `SSL_RSA_WITH_RC4_128_SHA`) however most of these ciphers are old/obsolete and I don't think we want to enable them by default anyways. If someone wants these older ciphers they should explicitly enable them instead of relying on the default (which in this case would mean none of the actual `SslUtils#DEFAULT_CIPHER_SUITES` ciphers are supported).